tests/src/Unit/OgAccessTestBase.php

```diff
@@ -73,6 +76,60 @@ public function setUp() {
     // This is for caching purposes only.
     $container->set('current_user', $this->user->reveal());
     \Drupal::setContainer($container);
+
+    $this->group = $this->groupEntity()->reveal();
+    $group_type_id = $this->group->getEntityTypeId();
+
+    $entity_id = 20;
+
+    // Set the Og::cache property values, to skip calculations.
+    $values = [];
+
+    $r = new \ReflectionClass('Drupal\og\Og');
+    $reflection_property = $r->getProperty('cache');
+    $reflection_property->setAccessible(TRUE);
+
+    // Mock the results of Og::getGroupIds().
+    $identifier = [
+      'Drupal\og\Og::getGroupIds',
+      $entity_id,
+      NULL,
+      NULL,
+    ];
+
+    $identifier = implode(':', $identifier);
+
+    $group_ids = [$group_type_id => [$this->group->id()]];
+    $values[$identifier] = $group_ids;
+
+    // Mock the results of Og::getUserMemberships().
+    $identifier = [
+      'Drupal\og\Og::getUserMemberships',
+      2,
+      OgMembershipInterface::STATE_ACTIVE,
+      // The field name.
+      NULL,
+    ];
+    $identifier = implode(':', $identifier);
+
+    // The cache is supposed to be holding the OG membership, however it is not
+    // used in the tests, so we just set a TRUE value.
+    $values[$identifier] = TRUE;
+
+    $reflection_property->setValue($values);
+
+    // Set the allowed permissions cache.
+    $r = new \ReflectionClass('Drupal\og\OgAccess');
+    $reflection_property = $r->getProperty('permissionsCache');
+    $reflection_property->setAccessible(TRUE);
+
+
+    $values = [];
+    foreach (['pre_alter', 'post_alter'] as $key) {
+      $values[$group_type_id][$this->group->id()][2][$key] = ['permissions' => ['update group']];
+    }
```

Comment:
there's the cache_context, but we don't bother setting it here. Mocking is such a lie! ;)